app/src/main/java/com/example/unlibrary/library/LibraryRepository.java

```diff
@@ -418,4 +425,126 @@ public void declineRequester(String requestedUID, String bookRequestedID, OnSucc
+                    } else {
+                        Log.e(TAG, "Error in fetching requests", requestTask.getException());
+                    }
+                });
+    }
+
+    /**
+     * Updates the handoff location for the book
+     *
+     * @param requestedUID      accepted requester user ID
+     * @param bookRequestedID   book ID request is associated with
+     * @param handoffLocation   LatLng of handoff location
+     * @param onSuccessListener code to call on success
+     * @param onFailureListener code to call on failure
+     */
+    public void updateHandoffLocation(String requestedUID, String bookRequestedID, LatLng handoffLocation, OnSuccessListener onSuccessListener, OnFailureListener onFailureListener) {
+        mDb.collection(REQUESTS_COLLECTION)
+                .whereEqualTo(REQUESTER, requestedUID)
+                .whereEqualTo(BOOK, bookRequestedID)
+                .whereNotEqualTo(STATE, ARCHIVED)
+                .get()
+                .addOnCompleteListener(task -> {
+                    if (task.isSuccessful()) {
+                        List<Request> requests = task.getResult().toObjects(Request.class);
+                        if (requests.size() != 1) {
+                            onFailureListener.onFailure(new Exception("Error updating handoff location"));
+                            return;
+                        }
+                        Request request = requests.get(0);
+                        mDb.collection(REQUESTS_COLLECTION).document(request.getId())
+                                .update(LOCATION, new GeoPoint(handoffLocation.latitude, handoffLocation.longitude))
+                                .addOnSuccessListener(onSuccessListener)
+                                .addOnFailureListener(onFailureListener);
+                    }
+                });
+    }
+
+    /**
+     * Fetches the handoff location for the selected book
+     *
+     * @param requestedUID      accepted requester user ID
+     * @param bookRequestedID   book ID request is associated with
+     * @param onFinished        code to call on success
+     * @param onFailureListener code to call on failure
+     */
+    public void fetchHandoffLocation(String requestedUID, String bookRequestedID, OnFinishedHandoffLocationListener onFinished, OnFailureListener onFailureListener) {
+        mDb.collection(REQUESTS_COLLECTION)
+                .whereEqualTo(BOOK, bookRequestedID)
+                .whereEqualTo(REQUESTER, requestedUID)
+                .whereNotEqualTo(STATE, ARCHIVED)
+                .get()
+                .addOnCompleteListener(task -> {
+                    if (task.isSuccessful()) {
+                        List<Request> requests = task.getResult().toObjects(Request.class);
+                        if (requests.size() != 1) {
+                            onFailureListener.onFailure(new Exception("Error fetching handoff location"));
```

Comment:
Same Nit: A more descriptive error string would be nice.

```suggestion
                            onFailureListener.onFailure(new Exception("Unexpected number of requests returned when fetching location.  " + requests.size() + " requests found."));
```